src/forgot-password/ForgotPasswordPage.jsx

```diff
@@ -50,27 +66,34 @@ const ForgotPasswordPage = (props) => {
 
   return (
     <Formik
-      onSubmit={(values) => {
-        if (values.isEmailValid) {
-          props.forgotPassword(values.email);
+      validate={(values) => {
+        const regex = new RegExp(VALID_EMAIL_REGEX, 'i');
+        if (!regex.test(values.email)) {
+          return { email: intl.formatMessage(messages['forgot.password.page.invalid.email.message']) };
         }
+        props.forgotPassword(values.email);
```

Comment:
Why we are calling the action in `validate` function? Can we keep this in `onSubmit`?